pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -151,5 +177,732 @@ func CreateCustomResourceDefinitions(clientset apiextensionsclient.Interface) er
+				},
+				"egress": {
+					Description: "Egress is a list of EgressRule which are enforced at egress. If " +
+						"omitted or empty, this rule does not apply at egress.",
+					Type: "array",
+					Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+						Schema: &apiextensionsv1beta1.JSONSchemaProps{
+							Ref: getStr("#/properties/EgressRule"),
+						},
+					},
+				},
+				"endpointSelector": {
+					Description: "EndpointSelector selects all endpoints which should be subject " +
+						"to this rule. Cannot be empty.",
+					Ref: getStr("#/properties/EndpointSelector"),
+				},
+				"ingress": {
+					Description: "Ingress is a list of IngressRule which are enforced at ingress. " +
+						"If omitted or empty, this rule does not apply at ingress.",
+					Type: "array",
+					Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+						Schema: &apiextensionsv1beta1.JSONSchemaProps{
+							Ref: getStr("#/properties/IngressRule"),
+						},
+					},
+				},
+				"labels": {
+					Description: "Labels is a list of optional strings which can be used to " +
+						"re-identify the rule or to store metadata. It is possible to lookup or " +
+						"delete strings based on labels. Labels are not required to be unique, " +
+						"multiple rules can have overlapping or identical labels.",
+					Type: "array",
+					Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+						Schema: &apiextensionsv1beta1.JSONSchemaProps{
+							Ref: getStr("#/properties/Label"),
+						},
+					},
+				},
+			},
+		},
+		"Service": {
+			Description: "Service wraps around selectors for services",
+			Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+				"k8sService": {
+					Description: "K8sService selects service by name and namespace pair",
+					Ref:         getStr("#/properties/K8sServiceNamespace"),
+				},
+				"k8sServiceSelector": {
+					Description: "K8sServiceSelector selects services by k8s labels. " +
+						"Not supported yet",
+					Ref: getStr("#/properties/ServiceSelector"),
+				},
+			},
+		},
+		"ServiceSelector": {
+			Description: "ServiceSelector is a label selector for k8s services",
+			Required: []string{
+				"LabelSelector",
+			},
+			Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+				"LabelSelector": {
+					Ref: getStr("#/properties/LabelSelector"),
+				},
+			},
+		},
+		"spec": {
+			Description: "Spec is the desired Cilium specific rule specification.",
+			Type:        "object",
+			Ref:         getStr("#/properties/Rule"),
+		},
+		"specs": {
+			Description: "Specs is a list of desired Cilium specific rule specification.",
+			Type:        "array",
+			Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+				Schema: &apiextensionsv1beta1.JSONSchemaProps{
+					Ref: getStr("#/properties/Rule"),
+				},
+			},
+		},
+		"status": {
+			Description: "Status is the status of the Cilium policy rule",
+			Ref:         getStr("#/properties/CiliumNetworkPolicyStatus"),
+		},
+		"Timestamp": {
```

Comment:
Do we need to validate timestamps? Are they ever used as input?

Reply:
Good point, I'll probably remove it a long with the `CiliumNetworkPolicyNodeStatus` and `CiliumNetworkPolicyStatus`